pages/api/hunks.ts

```diff
@@ -0,0 +1,14 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { saveHunk } from '../../utils/db/relevance';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const hunkjson = req.body;
+  console.log("Publishing message: ", hunkjson);
+  try {
+    saveHunk(hunkjson);
```

Comment:
Since `saveHunk` is an async function, I would suggest using the `.then` and `.catch` syntax of Promises than general try-catch.

```suggestion
    saveHunk(hunkjson).then(() => {
        res.status(200).send({ message: "Success" });
    })
    .catch(err => {
        console.error('Error publishing message:', error);
        res.status(500).json({ error: '<failure>Failed to publish message. Data must be in the form of a Buffer.</failure>' });
    });
```

Reply:
When I do this, it fails with the error status cannot be set after response is sent. What's the issue with using await instead? I have included try catch as well